flows/final_state_determination.py

```diff
@@ -0,0 +1,81 @@
+# /// script
+# dependencies = ["prefect>=3.0.0"]
+# ///
+
+"""
+This example demonstrates flow final state determination based on task states.
+"""
+
+import time
+
+from prefect import flow, task
+
+# Imports for type annotations
+from prefect.futures import PrefectFuture, PrefectFutureList
+
+
+def wait_for_futures():
+    """
+    Wait for all assigned PrefectFuture and PrefectFutureList objects in the calling function's local scope. If you do not assign the return value of a task to a variable, it will not be waited for.
+    """
+    import inspect
+
+    caller_locals = inspect.currentframe().f_back.f_locals
+
+    for item in caller_locals.values():
+        if isinstance(item, (PrefectFuture, PrefectFutureList)):
+            item.wait()
```

Comment:
hrm - can you explain the motivation for this? this strikes me as an extremely low level recommendation for most users

Reply:
Right, good observation. My intent was to create a utility that would identify and wait for futures automatically. This is the closest I could get on my own... in that it is limited in scope, but does what I want: wait for any futures.

The motivation for this particular example comes from talking to many users who have issues with two changes from 2.x -> 3.x: the changes in final state determination and the need to explicitly wait for submitted tasks.

Reply:
there's a `wait` utility in `prefect.futures`

Reply:
What's the advantage of using `wait` in `prefect.futures` in this scenario?

Reply:
I guess the complication here is that `locals()` might contain objects besides futures so I can't just pass them into `wait`.

Reply:
I'd say primarily that it is a first-class offering of the prefect library, it is what we recommend to "wait for futures"

Reply:
> I guess the complication here is that locals() might contain objects besides futures so I can't just pass them into wait.

maybe! i'd expect your `isinstance` check would catch that, but still I suspect this could be a subtle footgun bc of how its messing with low level python details. in my mind the big thing is that it undercuts ourselves if we have an offering specifically designed to be the public way to wait for futures